### PR TITLE
feat(flux2): Added ability to deploy extra K8s manifests

### LINKED ISSUE #71: [FR] Allow the Chart to create extra manifest
### Is your feature request related to a problem ?

Sometimes, accessory to the installation there may be manifest to be installed (for example a SecretProviderClass).

### Describe the solution you'd like.

It would be nice to be able to do this directly via the Chart using the approach that ArgoCD also uses.

Ref: https://github.com/argoproj/argo-helm/blob/cf57a71a154a1a8a19f5f9d3980d60eb3efd80c8/charts/argo-cd/values.yaml#L62-L87

### Describe alternatives you've considered.

I don't think there are alternatives

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by dwerder**
Signed-off-by: Daniel Werdermann <daniel.werdermann@gmail.com>

#### What this PR does / why we need it:

Sometimes, accessory to the installation there may be manifest to be installed (for example a SecretProviderClass)

#### Which issue this PR fixes
  - fixes #71 

#### Checklist

- [x] [DCO](https://github.com/fluxcd-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] helm-docs are updated
- [x] Helm chart is tested
- [x] Run `make reviewable`

### CODE REVIEW

**Review by haarchri — CHANGES_REQUESTED** [review]
LGTM - can you please add UnitTest?

File: charts/flux2/values.yaml [hunk 1]
```diff
@@ -222,3 +222,31 @@ watchallnamespaces: true
...
+  #         - objectName: "fluxcd"
+  #           objectType: "secretsmanager"
+  #           jmesPath:
+  #               - path: "client_id"
+  #                 objectAlias: "client_id"
+  #               - path: "client_secret"
+  #                 objectAlias: "client_secret"
+  #     secretObjects:
+  #     - data:
+  #       - key: client_id
```
Comment (stefanprodan) [inline charts/flux2/values.yaml:245]: Flux does not supports these key names for AWS auth. I'm guessing this example is for AWS S3?? If so the key names must be `accesskey` and `secretkey` (docs [here](https://fluxcd.io/docs/components/source/buckets/#static-authentication)).

Comment (dwerder) [reply charts/flux2/values.yaml:245]: You are right. I just used a random example. I will adjust it, to be more flux relevant.

**Review by haarchri — APPROVED** [review]
LGTM

**Pull request merged**